File: PHP/ProcessStringWithSpecialOperationsITest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit ProcessStringWithSpecialOperationsITest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("ProcessStringWithSpecialOperationsI.php");

class ProcessStringWithSpecialOperationsITest extends TestCase {

    public function testProcessStr() {
        $obj = new ProcessStringWithSpecialOperationsI();
        //Test 1
        $want = "ba";
        $got = $obj->processStr("a#b%*");
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = "";
        $got = $obj->processStr("z*#");
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/ProcessStringWithSpecialOperationsI_test.go
/**
 * @version GO 1.26.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestProcessStringWithSpecialOperationsI
 */
package GO

import (
	"reflect"
	"testing"
)

func TestProcessStringWithSpecialOperationsI(t *testing.T) {
	tables := []struct {
		want string
		s    string
	}{
		{"ba", "a#b%*"},
		{"", "z*#"},
	}

	for _, table := range tables {
		got := processStr(table.s)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}
